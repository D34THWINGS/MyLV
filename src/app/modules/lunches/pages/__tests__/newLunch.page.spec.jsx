import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import { NewLunchPage } from '../newLunch.page'

jest.unmock('../newLunch.page')

describe('NewLunchPage', () => {
  let props

  const getWrapper = () => shallow(<NewLunchPage {...props} />)

  beforeEach(() => {
    props = {
      postLunch: jest.fn(),
    }
  })

  it('should render new lunch form', () => {
    // When
    const wrapper = getWrapper()

    // Then
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('should pass render function to edit form', () => {
    // When
    const wrapper = shallow(getWrapper().prop('render')({ children: <div>Foo</div> }))

    // Then
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
