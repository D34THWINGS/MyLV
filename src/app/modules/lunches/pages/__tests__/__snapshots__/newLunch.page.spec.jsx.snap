// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewLunchPage should pass render function to edit form 1`] = `
<WithStyles(Paper)
  elevation={2}
>
  <WithStyles(CardContent)>
    <WithStyles(Typography)
      component="h2"
      gutterBottom={true}
      variant="headline"
    >
      Nouveau déjeuner
    </WithStyles(Typography)>
    <div>
      Foo
    </div>
  </WithStyles(CardContent)>
  <WithStyles(CardActions)>
    <WithStyles(Button)
      color="primary"
      disabled={true}
      size="small"
      type="submit"
    >
      Créer
    </WithStyles(Button)>
  </WithStyles(CardActions)>
</WithStyles(Paper)>
`;

exports[`NewLunchPage should render new lunch form 1`] = `
<WithStyles(ReduxForm)
  initialValues={
    Object {
      "attendees": Array [],
      "date": "2018-03-05",
    }
  }
  onFormSubmit={[MockFunction]}
  render={[Function]}
/>
`;
